gloo/docs/user_guides/gateway/tcp_proxy/_index.md

```diff
@@ -0,0 +1,194 @@
+---
+menuTitle: Tcp Proxy
+title: Gloo as a TCP proxy
+weight: 30
+description: Example of configuring Gloo as a L4/TCP proxy.
+---
+
+
+
+## Tcp Proxy
+
+In this tutorial, we'll take a look at using gloo as a TCP proxy. Envoy by default is an L4 proxy and is therefore
+more than up to the task. Gloo's TCP routing feature are slightly different than the rest of Gloo's routing as a result
+of the relative simplicity of TCP level routing. Current features include standard routing, ssl, and sni domain matching.
+
+### Resources 
+
+For reference on  the 
+
+- [Gateway]({{% ref "/v1/github.com/solo-io/gloo/projects/gateway/api/v1/gateway.proto.sk" %}})
+- [Proxy]({{% ref "/v1/github.com/solo-io/gloo/projects/gloo/api/v1/proxy.proto.sk" %}})
+
+### What you'll need
+
+You'll need to have Gloo installed on Kubernetes and have access to that Kubernetes cluster. Please refer to the
+[Gloo installation]({{< ref "/installation" >}}) for guidance on installing Gloo into Kubernetes.
+
+You'll also need access from the Kubernetes cluster to an external API. You will also need an accessible TCP service running 
+in cluster, this may be anything, for the purposes of this tutorial we will use a basic tcp-echo pod.
+
+Firstly apply the following yaml into the namespace of your choice. For the purposes of this tutorial we will be using `gloo-system`
+
+```yaml
+apiVersion: v1
+kind: Pod
+metadata:
+  labels:
+    gloo: tcp-echo
+  name: tcp-echo
+spec:
+  containers:
+  - image: soloio/tcp-echo:latest
+    imagePullPolicy: IfNotPresent
+    name: tcp-echo
+  restartPolicy: Always
+```
+```bash
+pbpaste | k apply -n gloo-system -f -
+```
+
+Once the `tcp-echo` pod is up and running we are ready to create our gateway resource and begin routing to it.
+
+As of vesion v2alpha1 of the [gateway]({{% ref "/v1/github.com/solo-io/gloo/projects/gateway/api/v1/gateway.proto.sk" %}}) 
+resource, it now supports 2 different types, those being HTTP, and TCP. 
+The [proxy]({{% ref "/v1/github.com/solo-io/gloo/projects/gloo/api/v1/proxy.proto.sk" %}}) resource has been extended as well with
+the TCP listener type. This is not a breaking change and therefore does not require an API upgrade. 
+
+The gateway will contain the following: 
+```yaml
+apiVersion: gateway.solo.io/v2alpha1
+kind: Gateway
+metadata:
+  name: tcp
+  namespace: gloo-system
+spec:
+  bindAddress: '::'
+  bindPort: 8000
+  tcpGateway:
+    destinations:
+    - name: one
+      destination:
+        single:
+          upstream:
+            name: gloo-system-tcp-echo-1025
```

Comment:
how is this get created if there's no service?

Reply:
Good point, need to add a service